### PR TITLE
refactor: [M3-6272] - MUI v5 Migration - `SRC > Features > Linodes` pt2

### PR DESCRIPTION
**Pull request by coliu-akamai**
## Description 📝
Part 2 of migrating the Linodes folder to MUI v5

## Major Changes 🔄
- made exports named (unless lazily imported)
- cleanup - remove React.FC, rename props to be more specific (only if they were being exported)
- converted some class components to functional components (note -- some of the more complicated ones like CreateLinodes.tsx left as is)
- removed unnecessary index files / updated respective imports

## How to test 🧪
1. Go to the Linodes pages and make sure functionality works as expected and that styling looks the same

### CODE REVIEW

File: packages/manager/src/features/Linodes/MigrateLinode/ConfigureForm.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ interface Props {
   selectedRegion: null | string;
 }
 
-const ConfigureForm = (props: Props) => {
+export const ConfigureForm = React.memo((props: Props) => {
```
Comment (tyler-akamai) [inline packages/manager/src/features/Linodes/MigrateLinode/ConfigureForm.tsx:20]: Rename `Props` --> `ConfigureFormProps`

File: packages/manager/src/features/Linodes/CloneLanding/Details.tsx [hunk 2]
```diff
@@ -275,11 +275,9 @@ const StyledHeader = styled('header', { label: 'StyledHeader' })(
   })
 );
 
-const StyledDiv = styled('div', { label: 'StyledDiv' })(({ theme }) => ({
+const StyledDiv = styled('div', { label: 'StyledDiv' })({
```
Comment (tyler-akamai) [inline packages/manager/src/features/Linodes/CloneLanding/Details.tsx:278]: You might want to make a separate styles file for Details.tsx

File: packages/manager/src/features/Linodes/LinodeEntityDetail.tsx [hunk 3]
```diff
@@ -166,51 +171,12 @@ export interface HeaderProps {
...
-    width: '100%',
-  },
-  chipWrapper: {
-    [theme.breakpoints.up('sm')]: {
-      '&.MuiGrid-item': {
-        marginTop: 2,
-      },
-    },
-  },
+// TODO: wasn't able to completely convert to styled components here due to chip styles not showing up
```
Comment (tyler-akamai) [inline packages/manager/src/features/Linodes/LinodeEntityDetail.tsx:174]: What do you think is causing it to not show up?

File: packages/manager/src/features/Linodes/LinodeEntityDetail.tsx [hunk 4]
```diff
@@ -166,51 +171,12 @@ export interface HeaderProps {
...
+const useHeaderStyles = makeStyles()((theme: Theme) => ({
   divider: {
     borderRight: `1px solid ${theme.borderColors.borderTypography}`,
     paddingRight: `16px !important`,
```
Comment (tyler-akamai) [inline packages/manager/src/features/Linodes/LinodeEntityDetail.tsx:178]: Do we need the `!important`?

File: packages/manager/src/features/Linodes/LinodeEntityDetail.style.ts [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+})(({ theme }) => ({
+  color: theme.textColors.headlineStatic,
+  fontFamily: theme.font.bold,
+}));
+
+export const StyledRightColumnGrid = styled(Grid, {
+  label: 'StyledRightColumnGrid',
+})(({ theme }) => ({
+  flexBasis: '75%',
+  flexWrap: 'nowrap',
```
Comment (tyler-akamai) [inline packages/manager/src/features/Linodes/LinodeEntityDetail.style.ts:61]: I don't think `flexWrap: 'nowrap',` adds anything.

File: packages/manager/src/features/Linodes/LinodeEntityDetail.style.ts [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+    fontFamily: '"UbuntuMono", monospace, sans-serif',
+    justifyContent: 'space-between',
+    padding: '4px 8px',
+    position: 'relative',
+  })
+);
+
+export const StyledCopyTooltip = styled(CopyTooltip, {
+  label: 'StyledCopyTooltip',
+})(({ theme }) => ({
```
Comment (tyler-akamai) [inline packages/manager/src/features/Linodes/LinodeEntityDetail.style.ts:142]: you can get rid of `theme`

File: packages/manager/src/features/Linodes/LinodeEntityDetail.style.ts [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+      width: 30,
+    },
+    overflowX: 'auto',
+    overflowY: 'hidden', // For Edge
+    paddingRight: 15,
+  })
+);
+
+export const StyledTableRow = styled(TableRow, { label: 'StyledTableRow' })(
+  ({ theme }) => ({
```
Comment (tyler-akamai) [inline packages/manager/src/features/Linodes/LinodeEntityDetail.style.ts:168]: same here

File: packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx [hunk 8]
```diff
@@ -17,7 +17,7 @@ import { Typography } from 'src/components/Typography';
 import { Paper } from 'src/components/Paper';
 import { resetEventsPolling } from 'src/eventsPolling';
 import { useMetadataCustomerTag } from 'src/features/Images/utils';
-import DiskSelect from 'src/features/Linodes/DiskSelect';
+import DiskSelect from 'src/features/Linodes/DiskSelect/DiskSelect';
```
Comment (abailly-akamai) [inline packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx:20]: No named export for for 'DiskSelect` ?

File: packages/manager/src/features/Linodes/CloneLanding/Configs.test.tsx [hunk 9]
```diff
@@ -4,11 +4,11 @@ import * as React from 'react';
 import { linodeConfigs } from 'src/__data__/linodeConfigs';
 import { wrapWithTheme } from 'src/utilities/testHelpers';
 
-import { Configs, Props } from './Configs';
+import Configs, { ConfigsProps } from './Configs';
```
Comment (abailly-akamai) [inline packages/manager/src/features/Linodes/CloneLanding/Configs.test.tsx:7]: You can separate those imports instead cause `ConfigsProps` should be an `import type { ConfigsProps } ...`

File: packages/manager/src/features/Linodes/CloneLanding/Configs.tsx [hunk 10]
```diff
@@ -12,13 +12,13 @@ import { TableRowEmpty } from 'src/components/TableRowEmpty/TableRowEmpty';
...
 import { ConfigSelection } from './utilities';
 
-export interface Props {
+export interface ConfigsProps {
   configSelection: ConfigSelection;
   configs: Config[];
   handleSelect: (id: number) => void;
 }
 
-export const Configs: React.FC<Props> = (props) => {
```
Comment (abailly-akamai) [inline packages/manager/src/features/Linodes/CloneLanding/Configs.tsx:21]: why did you remove the named export?

File: packages/manager/src/features/Linodes/CloneLanding/Disks.test.tsx [hunk 11]
```diff
@@ -4,13 +4,13 @@ import * as React from 'react';
 import { extDisk3, swapDisk } from 'src/__data__/disks';
 import { wrapWithTheme } from 'src/utilities/testHelpers';
 
-import { Disks, Props } from './Disks';
+import Disks, { DisksProps } from './Disks';
```
Comment (abailly-akamai) [inline packages/manager/src/features/Linodes/CloneLanding/Disks.test.tsx:7]: same this is an `import type ...`

File: packages/manager/src/features/Linodes/CloneLanding/Disks.tsx [hunk 12]
```diff
@@ -15,14 +15,14 @@ import { TableRowEmpty } from 'src/components/TableRowEmpty/TableRowEmpty';
...
-export interface Props {
+export interface DisksProps {
   diskSelection: DiskSelection;
   disks: Disk[];
   handleSelect: (id: number) => void;
   selectedConfigIds: number[];
 }
 
-export const Disks: React.FC<Props> = (props) => {
+const Disks = (props: DisksProps) => {
```
Comment (abailly-akamai) [inline packages/manager/src/features/Linodes/CloneLanding/Disks.tsx:25]: why removing the named export?

File: packages/manager/src/features/Linodes/LinodeEntityDetail.style.ts [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+import { Chip } from 'src/components/Chip';
+import { CopyTooltip } from 'src/components/CopyTooltip/CopyTooltip';
+import { TableCell } from 'src/components/TableCell';
+import { TableRow } from 'src/components/TableRow';
+
+// ---------------------------------------------------------------------
+// Header Styles
+// ---------------------------------------------------------------------
+
+export const StyledLink = styled(Link, { label: 'StyledLink' })(
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/features/Linodes/LinodeEntityDetail.style.ts:16]: Better to pass theme into each styled component rather than prop drilling IMO

File: packages/manager/src/features/Linodes/LinodeEntityDetail.style.ts [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+    marginLeft: theme.spacing(),
+  })
+);
+
+export const StyledChip = styled(Chip, { label: 'StyledChip' })(
+  ({ theme }) => ({
+    ...theme.applyStatusPillStyles,
+    borderRadius: 0,
+    fontSize: '0.875rem',
+    height: `24px !important`,
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/features/Linodes/LinodeEntityDetail.style.ts:32]: > ...theme.applyStatusPillStyles

I'd rather see a chip styled with an optional `pill` prop and have styles declared in the `Chip` component (not spread like this). We're trying to get away from using reusable styles that way

File: packages/manager/src/features/Linodes/LinodesDetail/LinodeSummary/LinodeSummary.tsx [hunk 15]
```diff
@@ -27,7 +27,7 @@ import {
   getMetrics,
 } from 'src/utilities/statMetrics';
 
-import NetworkGraph, { ChartProps } from './NetworkGraphs';
+import { NetworkGraphs, ChartProps } from './NetworkGraphs';
```
Comment (abailly-akamai) [inline packages/manager/src/features/Linodes/LinodesDetail/LinodeSummary/LinodeSummary.tsx:30]: `import type ...`

**Review by abailly-akamai — COMMENTED** [review]
@coliu-akamai nice work! Added some comments 

As a general comment this PR is just too big. I'd like to see it broken down so it's more reviewable and testable.

File: packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx (hunk 8 above)
Comment (coliu-akamai) [reply packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx:20]: oops got it!

File: packages/manager/src/features/Linodes/CloneLanding/Configs.tsx (hunk 10 above)
Comment (coliu-akamai) [reply packages/manager/src/features/Linodes/CloneLanding/Configs.tsx:21]: Ah I think I removed the named export because Configs as being lazily imported somewhere, so I kept the default export over the named one, but can put it back especially since Configs isn't lazily imported in the test file and wasn't originally used as a default export there

File: packages/manager/src/features/Linodes/LinodeEntityDetail.tsx (hunk 3 above)
Comment (coliu-akamai) [reply packages/manager/src/features/Linodes/LinodeEntityDetail.tsx:174]: good question, I've thoughts but no firm answers yet -- one of my initial thoughts was potentially because our Chip component is basically a wrapper around another StyledChip that already takes in props, and maybe putting another styled component on top of that was causing issues (but I have styled already styled components, so wasn't too sure)? I'd tried using the sx prop before as well, and that didn't work either (not sure if sx was clashing with the underlying styled component). Will look into it some more 👀

Comment (coliu-akamai) [reply packages/manager/src/features/Linodes/LinodeEntityDetail.tsx]: got it!

**Review by bnussman-akamai — APPROVED** [review]

**Review by abailly-akamai — APPROVED** [review]
Thanks for the changes! Looks good from what I could manually test

e2e test failure seems unrelated but I would recommend another run for confidence considering the size of this PR
